python/tests/test_modify_changelog.py

```diff
@@ -0,0 +1,300 @@
+    assert "Pull request number" in result.stdout
+    assert "Pull request title" in result.stdout
+
+
+def test_bump_version_patch():
+    """Test version bumping with patch increment."""
+    with tempfile.TemporaryDirectory() as temp_dir:
+        temp_path = Path(temp_dir)
+        
+        # Create mock Cargo.toml
+        cargo_content = '''[package]
+name = "test-package"
+version = "1.2.3"
+edition = "2021"
+'''
+        cargo_path = temp_path / "Cargo.toml"
+        cargo_path.write_text(cargo_content)
+        
+        # Create mock changelog
+        changelog_content = '''# Changelog
+
+## UNRELEASED
+
+## 1.2.3 (2024-01-01)
+
+- Some old change
+'''
+        docs_dir = temp_path / "docs"
+        docs_dir.mkdir()
+        changelog_path = docs_dir / "changelog.md"
+        changelog_path.write_text(changelog_content)
+        
+        # Run the script
+        result = subprocess.run([sys.executable, 
+                               str(Path(__file__).parent.parent.parent / "modify_changelog.py"), 
+                               "bump_version", "patch"], 
+                              capture_output=True, text=True, cwd=temp_path)
+        
+        assert result.returncode == 0
+        assert result.stdout.strip() == "1.2.4"
+        
+        # Check Cargo.toml was updated
+        updated_cargo = cargo_path.read_text()
+        assert 'version = "1.2.4"' in updated_cargo
+        
+        # Check changelog was updated
+        updated_changelog = changelog_path.read_text()
+        assert "## UNRELEASED" in updated_changelog
+        assert "## 1.2.4 (" in updated_changelog
+
+
+def test_bump_version_minor():
+    """Test version bumping with minor increment."""
+    with tempfile.TemporaryDirectory() as temp_dir:
+        temp_path = Path(temp_dir)
+        
+        # Create mock Cargo.toml
+        cargo_content = '''[package]
+name = "test-package"
+version = "1.2.3"
+edition = "2021"
+'''
+        cargo_path = temp_path / "Cargo.toml"
+        cargo_path.write_text(cargo_content)
+        
+        # Create mock changelog
+        changelog_content = '''# Changelog
+
+## UNRELEASED
+
+## 1.2.3 (2024-01-01)
+
+- Some old change
+'''
+        docs_dir = temp_path / "docs"
+        docs_dir.mkdir()
+        changelog_path = docs_dir / "changelog.md"
+        changelog_path.write_text(changelog_content)
+        
+        # Run the script
+        result = subprocess.run([sys.executable, 
+                               str(Path(__file__).parent.parent.parent / "modify_changelog.py"), 
+                               "bump_version", "minor"], 
+                              capture_output=True, text=True, cwd=temp_path)
+        
+        assert result.returncode == 0
+        assert result.stdout.strip() == "1.3.0"
+        
+        # Check Cargo.toml was updated
+        updated_cargo = cargo_path.read_text()
+        assert 'version = "1.3.0"' in updated_cargo
+
+
+def test_bump_version_major():
```

Comment:
Can you combine these to one test using pytest.param to pass in the start version, end version, and bump type?